### DISCUSSION TITLE
Experience with multiprocessing

### DISCUSSION THREAD

**Discussion by MatteoVoges** [omry/omegaconf#1116] [Q&A]
Hey,

I'm struggling a bit with using multiprocessing with OmegaConf.
Because it's not officially supported I don't want to declare some errors as bugs, but maybe you know some workarounds.

Currently my problems are, that some custom resolvers are not registered or even simple interpolations can't get resolved.

**Comment by odelalleau** [comment]
multiprocesssing should work fine, multithreading is what may cause issues if you're not careful (OmegaConf is not thread-safe)

My general advice with multiprocessing (not OmegaConf-specific) is to use the "spawn" start method (for cross-platform compatibility and better understanding of what happens -- at least for me) and ensure all modules are imported when loading the target.

Feel free to open specific issues.
